datafusion/src/physical_plan/datetime_expressions.rs

```diff
@@ -203,6 +203,10 @@ fn date_trunc_single(granularity: &str, value: i64) -> Result<i64> {
             .and_then(|d| d.with_minute(0))
             .and_then(|d| d.with_hour(0))
             .and_then(|d| d.with_day0(0)),
+        "quarter" => value
```

Comment:
If I understand rightly, this means three months?  So you also need `.and_then(|d| d.with_day0(0))`?

Reply:
I don't fully understand how this code will truncate the timestamp to the calendar quarter. It seems like it will truncate to "day".

I have suggested some tests that would show its behavior one way or the other